guacamole-common-js/src/main/webapp/modules/SessionRecording.js

```diff
@@ -872,6 +911,19 @@ Guacamole.SessionRecording = function SessionRecording(source, refreshInterval)
      */
     this.onpause = null;
 
+    /**
+     * Fired whenever a new batch of typed text extracted from key events
+     * is available.
+     *
+     * @event
+     * @param {!String} text
+     *     The typed text associated with the batch of text.
+     *
+     * @param {!number} timestamp
+     *     The relative timestamp associated with the batch of text.
```

Comment:
Could this event should be richer? There definitely needs to be the rough text equivalent available for easy consumption, so this much is pretty good, but:

* This doesn't appear to provide a way for consuming code to distinguish between non-printable keypresses (like "<Backspace">) and the text equivalent to how we represent them (typing "<", "B", "a", ... etc.), which would prevent implementations from styling such keys differently or taking reliable automated action based on the presumed text typed.
* Timing information for keypresses vs. frames is lost (ie: there's no way for consuming code to choose to represent the keys pressed/released one key at a time)

Perhaps there's a way we can represent that in an object received by this event?

Reply:
This is addressed now.

Reply:
This is addressed now.